### PR TITLE
make use of new SAM api that return resource information when creating resources

### PR DESCRIPTION
**Pull request by kcibul**
See https://broadworkbench.atlassian.net/browse/AS-55

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/model/SamModel.scala [hunk 1]
```diff
@@ -110,18 +110,29 @@ object SamWorkflowCollectionPolicyNames {
...
-case class SamResourceWithPolicies(resourceId: String, policies: Map[SamResourcePolicyName, SamPolicy], authDomain: Set[String])
+case class SamResourceWithPolicies(resourceId: String, policies: Map[SamResourcePolicyName, SamPolicy], authDomain: Set[String], returnResource: Boolean = false)
 case class SamResourceIdWithPolicyName(resourceId: String, accessPolicyName: SamResourcePolicyName, authDomainGroups: Set[WorkbenchGroupName], missingAuthDomainGroups: Set[WorkbenchGroupName], public: Boolean)
 case class SamPolicySyncStatus(lastSyncDate: String, email: WorkbenchEmail)
 
+case class SamCreateResourceResponse(resourceTypeName: String, resourceId: String, authDomain: Set[String], accessPolicies: Set[SamCreateResourcePolicyResponse])
+case class SamCreateResourcePolicyResponse(id: SamCreateResourceAccessPolicyIdResponse, email: String)
+case class SamCreateResourceAccessPolicyIdResponse(accessPolicyName: String, resource: SamCreateResourceAccessPolicyIdResourceIdResponse)
+case class SamCreateResourceAccessPolicyIdResourceIdResponse(resourceId: String, resourceTypeName: String)
```
Comment (davidangb) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/model/SamModel.scala:120]: what a name!

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 2]
```diff
@@ -1987,24 +1985,33 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
...
+              if (policyName == SamWorkspacePolicyNames.projectOwner && workspaceRequest.authorizationDomain.getOrElse(Set.empty).isEmpty) {
+                // when there isn't an auth domain, we will use the billing project admin policy email directly on workspace
+                // resources instead of synching an extra group. This helps to keep the number of google groups a user is in below
+                // the limit of 2000
+                Future.successful(())
+              } else if (WorkspaceAccessLevels.withPolicyName(policyName.value).isDefined) {
+                // only sync policies that have corresponding WorkspaceAccessLevels to google because only those are
+                // granted bucket access (and thus need a google group)
+                samDAO.syncPolicyToGoogle(SamResourceTypeNames.workspace, workspaceId, policyName)
+                Future.successful(())
```
Comment (davidangb) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala]: this looks like a functional change. Previously this code path returned the value of `syncPolicyToGoogle` ... now it is hardcoded to be successful. Did you intend this?

**Review by davidangb — COMMENTED** [review]
one question, otherwise looks great!

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala (hunk 2 above)
Comment (kcibul) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala]: no -- excellent catch!!

**Review by davidangb — APPROVED** [review]
:+1: once tests pass

**Comment by kcibul** [comment]
jenkins retest

**Comment by kcibul** [comment]
jenkins retest

**Pull request merged**